internal/build/cache_builder.go

```diff
@@ -0,0 +1,125 @@
+package build
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"strings"
+
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+	"github.com/windmilleng/tilt/internal/docker"
+	"github.com/windmilleng/tilt/internal/dockerfile"
+	"github.com/windmilleng/tilt/internal/logger"
+)
+
+// Cache directories are stored at the same image name, but with just the cachedir
+// contents and at a particular tag.
+const CacheTagPrefix = "tilt-cache-"
```

Comment:
Should this have a version number in case we change the format in the future?

Reply:
sure! tho in that case we could just add a version number (and/or bake the version number into the md5 hash)